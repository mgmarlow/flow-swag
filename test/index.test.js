const generateFlowTypes = require('../index.js')

describe('#generateFlowTypes', () => {
  describe('alternative .prettierc', () => {
    const options = {
      prettierConfig: './test/swagger/.prettierrc',
    }

    it('should generate flow types', () => {
      const source = './test/swagger/pet_store.yaml'

      const types = generateFlowTypes(source, options)
      expect(types).toMatchSnapshot()
    })
  })

  describe('.yaml source', () => {
    describe('camelizeKeys is true', () => {
      const options = {
        camelizeKeys: true,
      }

      it('should generate flow types', () => {
        const source = './test/swagger/pet_store.yaml'

        const types = generateFlowTypes(source, options)
        expect(types).toMatchSnapshot()
      })
    })

    describe('camelizeKeys is false', () => {
      const options = {
        camelizeKeys: false,
      }

      it('should generate flow types', () => {
        const source = './test/swagger/pet_store.yaml'

        const types = generateFlowTypes(source, options)
        expect(types).toMatchSnapshot()
      })
    })
  })

  describe('.json source', () => {
    describe('camelizeKeys is true', () => {
      const options = {
        camelizeKeys: true,
      }

      it('should generate flow types', () => {
        const source = './test/swagger/pet_store.json'

        const types = generateFlowTypes(source, options)
        expect(types).toMatchSnapshot()
      })
    })

    describe('camelizeKeys is false', () => {
      const options = {
        camelizeKeys: false,
      }

      it('should generate flow types', () => {
        const source = './test/swagger/pet_store.json'

        const types = generateFlowTypes(source, options)
        expect(types).toMatchSnapshot()
      })
    })
  })

  describe('swagger v2', () => {
    it('should generate flow types', () => {
      const source = './test/swagger/pet_store_v2.json'

      const types = generateFlowTypes(source, {})
      expect(types).toMatchSnapshot()
    })
  })
})
