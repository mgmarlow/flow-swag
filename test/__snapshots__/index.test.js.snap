// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#generateFlowTypes .json source camelizeKeys is false should generate flow types 1`] = `
"// @flow
export type Order = {
  id: number,
  pet_id: number,
  quantity: number,
  ship_date: string,
  status: 'placed' | 'approved' | 'delivered',
  complete: boolean,
}

export type Category = {
  id: number,
  name: string,
}

export type User = {
  id: number,
  username: string,
  first_name: string,
  last_name: string,
  email: string,
  password: string,
  phone: string,
  user_status: number,
}

export type Tag = {
  id: number,
  name: string,
}

export type Pet = {
  id: number,
  category: Category,
  name: string,
  photo_urls: string[],
  tags: Tag[],
  status: 'available' | 'pending' | 'sold',
}

export type ApiResponse = {
  code: number,
  type: string,
  message: string,
}
"
`;

exports[`#generateFlowTypes .json source camelizeKeys is true should generate flow types 1`] = `
"// @flow
export type Order = {
  id: number,
  petId: number,
  quantity: number,
  shipDate: string,
  status: 'placed' | 'approved' | 'delivered',
  complete: boolean,
}

export type Category = {
  id: number,
  name: string,
}

export type User = {
  id: number,
  username: string,
  firstName: string,
  lastName: string,
  email: string,
  password: string,
  phone: string,
  userStatus: number,
}

export type Tag = {
  id: number,
  name: string,
}

export type Pet = {
  id: number,
  category: Category,
  name: string,
  photoUrls: string[],
  tags: Tag[],
  status: 'available' | 'pending' | 'sold',
}

export type ApiResponse = {
  code: number,
  type: string,
  message: string,
}
"
`;

exports[`#generateFlowTypes .yaml source camelizeKeys is false should generate flow types 1`] = `
"// @flow
export type Order = {
  id: number,
  object_test: {
    id: number,
    name: string,
  },
  pet_id: number,
  quantity: number,
  ship_date: string,
  status: 'placed' | 'approved' | 'delivered',
  complete: boolean,
}

export type Category = {
  id: number,
  name: string,
}

export type User = {
  id: number,
  username: string,
  first_name: string,
  last_name: string,
  email: string,
  password: string,
  phone: string,
  user_status: number,
}

export type Tag = {
  id: number,
  name: string,
}

export type Pet = {
  id: number,
  category: Category,
  name: string,
  photo_urls: string[],
  tags: Tag[],
  status: 'available' | 'pending' | 'sold',
}

export type ApiResponse = {
  code: number,
  type: string,
  message: string,
}
"
`;

exports[`#generateFlowTypes .yaml source camelizeKeys is true should generate flow types 1`] = `
"// @flow
export type Order = {
  id: number,
  objectTest: {
    id: number,
    name: string,
  },
  petId: number,
  quantity: number,
  shipDate: string,
  status: 'placed' | 'approved' | 'delivered',
  complete: boolean,
}

export type Category = {
  id: number,
  name: string,
}

export type User = {
  id: number,
  username: string,
  firstName: string,
  lastName: string,
  email: string,
  password: string,
  phone: string,
  userStatus: number,
}

export type Tag = {
  id: number,
  name: string,
}

export type Pet = {
  id: number,
  category: Category,
  name: string,
  photoUrls: string[],
  tags: Tag[],
  status: 'available' | 'pending' | 'sold',
}

export type ApiResponse = {
  code: number,
  type: string,
  message: string,
}
"
`;

exports[`#generateFlowTypes alternative .prettierc should generate flow types 1`] = `
"// @flow
export type Order = {
  id: number,
  object_test: {
    id: number,
    name: string
  },
  pet_id: number,
  quantity: number,
  ship_date: string,
  status: \\"placed\\" | \\"approved\\" | \\"delivered\\",
  complete: boolean
};

export type Category = {
  id: number,
  name: string
};

export type User = {
  id: number,
  username: string,
  first_name: string,
  last_name: string,
  email: string,
  password: string,
  phone: string,
  user_status: number
};

export type Tag = {
  id: number,
  name: string
};

export type Pet = {
  id: number,
  category: Category,
  name: string,
  photo_urls: string[],
  tags: Tag[],
  status: \\"available\\" | \\"pending\\" | \\"sold\\"
};

export type ApiResponse = {
  code: number,
  type: string,
  message: string
};
"
`;

exports[`#generateFlowTypes swagger v2 should generate flow types 1`] = `
"// @flow
export type Order = {
  id: number,
  petId: number,
  quantity: number,
  shipDate: string,
  status: 'placed' | 'approved' | 'delivered',
  complete: boolean,
}

export type User = {
  id: number,
  username: string,
  firstName: string,
  lastName: string,
  email: string,
  password: string,
  phone: ?string,
  userStatus: number,
}

export type Category = {
  id: number,
  name: string,
}

export type Tag = {
  id: number,
  name: string,
}

export type Pet = {
  id: number,
  category: Category,
  name: string,
  photoUrls: ?(string[]),
  tags: ?(Tag[]),
  status: 'available' | 'pending' | 'sold',
}

export type ApiResponse = {
  code: number,
  type: string,
  message: string,
}
"
`;
